File: 1302-delete-characters-to-make-fancy-string/delete-characters-to-make-fancy-string.java
class Solution {
    public String makeFancyString(String s) {
        int streak = 1;
        StringBuilder result = new StringBuilder();
        for (int i = 0; i < s.length(); i++){
            if (i > 1 && s.charAt(i) == result.charAt(result.length() - 1) && result.charAt(result.length() - 1) == result.charAt(result.length() - 2)) continue;
            result.append(s.charAt(i));
        }
        return result.toString();
    }
}

File: 1302-delete-characters-to-make-fancy-string/delete-characters-to-make-fancy-string.cpp
class Solution {
public:
    string makeFancyString(string s) {
        int streak = 1;
        string result = "";
        for (int i = 0; i < s.length(); i++){
            if (i > 0 && s[i] == s[i-1]) streak += 1;
            else streak = 1;
            if (streak < 3) result += s[i];
        }
        return result;
    }
};